invenio_rdm_records/services/components/internal_notes.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM service component for custom fields."""
+from copy import copy, deepcopy
+from datetime import datetime, timezone
+
+from invenio_pidstore.providers.recordid_v2 import RecordIdProviderV2
+
+from .metadata import MetadataComponent
+
+
+class InternalNotesComponent(MetadataComponent):
+    """Service component for custom fields."""
+
+    field = "internal_notes"
+    new_version_skip_fields = []
+
+    def create(self, identity, data=None, record=None, **kwargs):
+        """Inject note to the record."""
+        notes = data.get(self.field, [])
+        for note in notes:
+            note.update({
+                "added_by": {"user": identity.id},
+                "timestamp": datetime.now(timezone.utc).isoformat(),
+                "id": RecordIdProviderV2.generate_id()
+            })
+        record.update({"internal_notes": notes})
+
+    def update_draft(self, identity, data=None, record=None, **kwargs):
+        """Inject parsed metadata to the record."""
+        notes_updated = data.get(self.field, [])
+        notes = deepcopy(record.get(self.field, []))
+        existing_ids = set([note["id"] for note in notes])
+        new_ids = set([note["id"] for note in notes_updated])
+        to_delete = existing_ids - new_ids
+
+        if sorted(existing_ids) != sorted(new_ids):
+            for note in notes_updated:
+                if note["id"] in existing_ids:
```

Comment:
I understand that the new list is of:
```
new_notes = [{id: 'existing1', note: 'blabla'}, {note: 'new note'}, ....]
```

if the above is correct then we cannot assume that `note['id']` is valid for the new notes.